modules/tftp/server.rb

```diff
@@ -150,6 +150,36 @@ def pxeconfig_file(mac)
     end
   end
 
+  def self.fetch_boot_image(image_dst, url, files)
+    # Verify image_dst does not contain ".." (switch folder)
+    raise "TFTP image file extraction error: #{file_path}" unless image_dst.split("..").size <= 1
+
+    base_path = Pathname.new("/var/www/html/pub/installation_media")
+    image_path = Pathname.new(File.expand_path(image_dst, base_path)).cleanpath
+    extr_image_dir = Pathname.new(image_path.to_s.delete_suffix(".iso"))
```

Comment:
This assumes the suffix is `.iso`. How about using `Pathname`:
```suggestion
    extr_image_dir = image_path.sub_ext('')
```

Reply:
Done.